src/VCAuthn/ACAPy/ACAPYClient.cs

```diff
@@ -0,0 +1,66 @@
+using System;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+
+namespace VCAuthn.ACAPY
+{
+    public interface IACAPYClient
+    {
+        Task<WalletPublicDid> WalletDidPublic();
+        string GetServicePublicUrl();
+    }
+
+    public class ACAPYClient : IACAPYClient
+    {
+        private readonly ILogger<ACAPYClient> _logger;
+        private readonly string _baseUrl;
+        private HttpClient _httpClient;
+        
+        public const string WalletDidPublicUri = "/wallet/did/public";
+
+        public ACAPYClient(IConfiguration config, ILogger<ACAPYClient> logger)
+        {
+            _httpClient = new HttpClient();
+            _logger = logger;
+            _baseUrl = config.GetValue<string>("BaseUrl");
+        }
+
+        public string GetServicePublicUrl()
+        {
+            return _baseUrl;
+        }
+
+        public async Task<WalletPublicDid> WalletDidPublic()
+        {
+            var request = new HttpRequestMessage
+            {
+                Method = HttpMethod.Get,
+                RequestUri = new Uri($"{_baseUrl}{WalletDidPublicUri}")
+            };
+
+            try
+            {
+                var response = await _httpClient.SendAsync(request);
+                var responseContent = await response.Content.ReadAsStringAsync();
+
+                _logger.LogDebug($"Status: [{response.StatusCode}], Content: [{responseContent}, Headers: [{response.Headers}]");
+
+                switch (response.StatusCode)
```

Comment:
Could we remove the need for a switch here and check for a success status code?

```csharp
if (!response.IsSuccessStatusCode) {
   throw new Exception($"Wallet Did public request error. Code: {response.StatusCode}");
}
return JsonConvert.DeserializeObject<WalletDidPublicResponse>(responseContent).Result;
```

Reply:
sorry, don't see why is it better

Reply:
I think `if` increases readability and fits the program flow better.

A switch, implies that we are, or will be, handling multiple cases.

Using `if` means we are affirming a single test and we do not have multiple possible conditions.

I think that `if` fits this use case as we are affirming a single test
Functionally they are the same